packages/odyssey-react-mui/src/labs/DataComponents/DataView.tsx

```diff
@@ -0,0 +1,388 @@
+import {
+  availableLayouts as allAvailableLayouts,
+  densityValues,
+} from "./constants";
+import { Box } from "../../Box";
+import { DataFilter, DataFilters } from "../DataFilters";
+import { useFilterConversion } from "./useFilterConversion";
+import { BulkActionMenu } from "./BulkActionsMenu";
+import { LayoutSwitcher } from "./LayoutSwitcher";
+import { TableSettings } from "./TableSettings";
+import { Callout } from "../../Callout";
+import { Pagination, usePagination } from "../../Pagination";
+import { t } from "i18next";
+import { fetchData } from "./fetchData";
+import { TableContent } from "./TableContent";
+import { StackContent } from "./StackContent";
+import { useRowReordering } from "../../DataTable/useRowReordering";
+import { EmptyState } from "../../EmptyState";
+import {
+  MRT_Row,
+  MRT_RowData,
+  MRT_RowSelectionState,
+} from "material-react-table";
+
+export type DataViewProps = UniversalProps & ViewProps<Layout>;
+
+const DataView = ({
+  getData: getDataFn,
+  getRowId: getRowIdProp,
+  availableLayouts = allAvailableLayouts,
+  initialLayout,
+  hasSearch,
+  hasFilters,
+  hasSearchSubmitButton,
+  searchDelayTime,
+  bulkActionMenuItems,
+  hasPagination,
+  paginationType = "paged",
+  resultsPerPage = 20,
+  currentPage = 1,
+  totalRows,
+  errorMessage: errorMessageProp,
+  hasRowReordering,
+  isRowReorderingDisabled,
+  onReorderRows,
+  emptyPlaceholder,
+  noResultsPlaceholder,
+  tableOptions,
+  stackOptions,
+  hasRowSelection,
+  onChangeRowSelection,
+  isEmpty: isEmptyProp,
+  isLoading: isLoadingProp,
+  isNoResults: isNoResultsProp,
+}: DataViewProps) => {
+  const [currentLayout, setCurrentLayout] = useState<Layout>(
+    initialLayout ?? availableLayouts[0],
+  );
+
+  const [data, setData] = useState<MRT_RowData[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(isLoadingProp ?? true);
+  const [isEmpty, setIsEmpty] = useState<boolean>(isEmptyProp ?? true);
+  const [isNoResults, setIsNoResults] = useState<boolean>(
+    isNoResultsProp ?? false,
+  );
+  const [errorMessage, setErrorMessage] =
+    useState<UniversalProps["errorMessage"]>(errorMessageProp);
+
+  const [search, setSearch] = useState<string>("");
+  const [initialFilters, setInitialFilters] = useState<DataFilter[]>();
+  const [filters, setFilters] = useState<DataFilter[]>();
+
+  const [draggingRow, setDraggingRow] = useState<MRT_Row<MRT_RowData> | null>();
+
+  const [rowSelection, setRowSelection] = useState<MRT_RowSelectionState>({});
+
+  useEffect(() => {
+    onChangeRowSelection?.(rowSelection);
+  }, [rowSelection, onChangeRowSelection]);
+
+  const [pagination, setPagination] = useState({
+    pageIndex: currentPage,
+    pageSize: resultsPerPage,
+  });
+
+  const [tableState, setTableState] = useState<TableState>({
+    rowDensity: tableOptions?.initialDensity ?? densityValues[0],
+    columnSorting: [],
+    columnVisibility: {},
+  });
+
+  const hasMultipleAvailableLayouts = useMemo(
```

Comment:
I thought `availableLayouts` was always going to be an array now? Can remove the string check if that is the case